src/Widgets/DisplayWidget.vala

```diff
@@ -205,7 +205,9 @@ public class Display.DisplayWidget : Gtk.EventBox {
         destroy.connect (() => display_window.destroy ());
 
         use_switch.notify["active"].connect (() => {
-            //output_info.set_active (use_switch.active);
+            if (!virtual_monitor.set_active (use_switch.active)) {
```

Comment:
You should bind the `use_switch.active` "sensitive" property to `virtual_monitor.is_mirrored` with an `INVERT_BOOLEAN` binding flag.  That way the user would not be able to turn off a mirrored monitor and no boolean setter is needed.

Reply:
That worked great, thanks!